### ISSUE TITLE
Error in the machine learning pipeline illustration?

### ISSUE THREAD

**Issue by viggotw** [APMonitor/pds#1]
I just reviewed your illustration of the classical ML pipeline, and I was surprised that you have put the splitting into training- validation- and testing-set so far at the end. My experience is that one always would like to "hide" the test set as soon as possible, ensuring that it never influences any decisions up until the final testing and scoring of your model(s). Scaling and feature engineering should definitely be done without access to the test set, and arguably even the EDA-step.

![](https://apmonitor.com/pds/uploads/Main/pds_overview.png)

**Comment by APMonitor** [comment]
Thanks for sharing your experience and the insightful comments. You are correct that Test data should be created and curated independently so that it is representative of what you would observe during deployment. The Train and Validation data sets certainly belong near the end just prior to the learning algorithms. The Test data would need to have the same scaling as the training sets. The Test data also has some type of gross error detection and cleansing to eliminate outliers. If additional features are created for the Training and Validation then they will also need to be created for the Test set. A more correct chart would have a reduced data preparation flow chart after the Learning to curate the Test data.
